http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzureInterpreter.scala

```diff
@@ -0,0 +1,394 @@
+
+      createAzureRuntimeResult <- createRuntime(params)
+
+      wsmWorkspace <- wsmDao.getWorkspace(msg.workspaceId)
+      _ <- monitorCreateRuntime(
+        PollRuntimeParams(msg.workspaceId,
+                          runtime,
+                          wsmWorkspace.azureContext.managedResourceGroupName,
+                          createAzureRuntimeResult.jobReport.id)
+      )
+    } yield ()
+
+  /** Creates an Azure VM but doesn't wait for its completion.
+   * This includes creation of all child Azure resources (disk, network, ip), and assumes these are created synchronously
+   * */
+  private def createRuntime(params: CreateAzureRuntimeParams)(implicit ev: Ask[F, AppContext]): F[CreateVmResult] =
+    for {
+      createIpResp <- createIp(params)
+      createDiskResp <- createDisk(params)
+      createNetworkResp <- createNetwork(params)
+
+      vmCommon = getCommonFields(ControlledResourceName(params.runtime.runtimeName.asString),
+                                 config.vmControlledResourceDesc,
+                                 params.runtime.auditInfo.creator)
+      vmRequest: CreateVmRequest = CreateVmRequest(
+        params.workspaceId,
+        vmCommon,
+        CreateVmRequestData(
+          params.runtime.runtimeName,
+          params.runtimeConfig.region,
+          VirtualMachineSizeTypes.fromString(params.runtimeConfig.machineType.value),
+          params.vmImage,
+          createIpResp.resourceId,
+          createDiskResp.resourceId,
+          createNetworkResp.resourceId
+        )
+      )
+      createVmResp <- wsmDao.createVm(vmRequest)
+      _ <- dbRef.inTransaction(
+        controlledResourceQuery.save(
+          params.runtime.id,
+          createVmResp.vm.resourceId,
+          WsmResourceType.AzureVm,
+          params.pd.name.value
+        )
+      )
+    } yield createVmResp
+
+  private def monitorCreateRuntime(params: PollRuntimeParams)(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    implicit val azureRuntimeCreatingDoneCheckable: DoneCheckable[VirtualMachine] = (v: VirtualMachine) =>
+      v.powerState().toString.equals(PowerState.RUNNING.toString)
+    implicit val wsmCreateVmDoneCheckable: DoneCheckable[CreateVmResult] = (v: CreateVmResult) =>
+      v.jobReport.status.equals(WsmJobStatus.Succeeded)
+    for {
+      ctx <- ev.ask
+
+      getWsmJobResult = wsmDao.getCreateVmJobResult(GetJobResultRequest(params.workspaceId, params.jobId))
+      getRuntimeOpt <- azureComputeManager.getAzureVm(params.runtime.runtimeName, params.resourceGroup)
+
+      getRuntime = F.fromOption(
+        getRuntimeOpt,
+        PubsubHandleMessageError.AzureRuntimeError(params.runtime.id,
+                                                   ctx.traceId,
+                                                   None,
+                                                   "Could not retrieve vm controlled resource for runtime from DB")
+      )
```

Comment:
I don't think this is what you want. line 170 will be executed immediately and only that once...it won't be executed repeated in the stream below becuz `getRuntime` will be a static value instead of the action of retrieving the result

it should be
```
getRuntime = azureComputeManager.getAzureVm(params.runtime.runtimeName, params.resourceGroup).flatMap(op => F.fromOptoin(op))
```

Reply:
Ah thanks, the proper logic was lost when I refactored it to return an option